lib/curlx/fopen.c

```diff
@@ -181,15 +183,17 @@ static bool fix_excessive_path(const TCHAR *in, TCHAR **out)
 
 #ifndef _UNICODE
   /* convert unicode full path to multibyte output */
-  needed = wcstombs(NULL, fbuf, 0);
-  if(needed == (size_t)-1 || needed >= max_path_len)
+  if(wcstombs_s(&needed, NULL, 0, fbuf, 0))
+    goto cleanup;
+  if(needed >= max_path_len)
     goto cleanup;
   ++needed; /* for NUL */
```

Comment:
```suggestion
```
the wcstombs_s version includes the null terminator in needed multibyte characters, unlike wcstombs